Classes/Utility/CustomErrorPageUtility.php

```diff
@@ -142,7 +142,30 @@ private function findErrorPage(string $currentUrl, array $configuration, int $ty
         }
 
         foreach ($configurationAllocations as $regex => $url) {
-            if (preg_match($regex, $currentUrl)) {
+            if (preg_match($regex, $params['currentUrl'])) {
+                try {
+                    $GLOBALS['TSFE']->id = $GLOBALS['TSFE']->domainStartPage;
+                    $GLOBALS['TSFE']->getPageAndRootline();
+                    $GLOBALS['TSFE']->initTemplate();
+                    $GLOBALS['TSFE']->tmpl->start($GLOBALS['TSFE']->rootLine);
+                    $GLOBALS['TSFE']->getConfigArray();
+                    $GLOBALS['TSFE']->settingLanguage();
+                    $GLOBALS['TSFE']->calculateLinkVars();
+
+                    if ($GLOBALS['TSFE']->linkVars) {
+                        $url .= $GLOBALS['TSFE']->linkVars;
+                    }
+                } catch (\Exception $e) {
+                    $message = 'Could not build a localized "pageNotFound" that belongs to this domaine. '
+                        . "(pid: {$GLOBALS['TSFE']->domainStartPage})";
+                    $this->logger->critical($message);
+                }
+
+                if ($configuration[$hostName]['forward-reason'] === true
+                    || $configuration['_DEFAULT']['forward-reason'] === true) {
+                    $url .= '&reason=' . urlencode($params['reasonText']);
```

Comment:
Is the variable $params['reasonText'] always set? If not, you could add an !empty check before.

Reply:
Check added.